### PR TITLE
feat : disable gov, staking and transers tabs with authz

### PR DESCRIPTION
**Pull request by Teja2045**
tried to implement useEffect with dependency[authzState] directly inside home page, but its forming an infinite loop. Tried many implementations but most had same kind of problems.
This was the only implementation I could find to resolve the loop issue

### CODE REVIEW

**Comment by Teja2045** [comment]
https://github.com/vitwit/resolute/blob/d7d4f7a228e4742b0f8d98c8fd524e6a9057e5f3/frontend/src/features/authz/authzSlice.js#L19-L28

added the new field to reduce the number of renders (useEffects in customAppBar) when authz tabs state changes

File: frontend/src/utils/authorizations.ts [hunk 1]
```diff
@@ -214,3 +228,57 @@ export function getMsgNameFromAuthz(authorization: any): string {
...
+      switch (authorizations[i].authorization?.msg) {
+        case SEND_V1BETA1_TYPEURL:
+          result.sendEnabled = true;
+          break;
+
+        case VOTE_V1BETA1_TYPEURL:
+          result.govEnabled = true;
+          break;
+
+        case WITHDRAW_REWARDS_V1BETA1_TYPEURL:
```
Comment (aleem1314) [inline frontend/src/utils/authorizations.ts:269]: Also add a new `else if` branch for `STAKE_AUTHORIZATION`.

File: frontend/src/utils/authorizations.ts [hunk 2]
```diff
@@ -214,3 +228,57 @@ export function getMsgNameFromAuthz(authorization: any): string {
...
+        case DELEGATE_V1BETA1_TYPEURL:
+        case UNDELEGATE_V1BETA1_TYPEURL:
+        case REDELEGATE_V1BETA1_TYPEURL:
+          result.stakingEnabled = true;
+          break;
+
+        case GRANT_ALLOWANCE_V1BETA1_TYPEURL:
+        case REVOKE_ALLOWANCE_V1BETA1_TYPEURL:
```
Comment (aleem1314) [inline frontend/src/utils/authorizations.ts:277]: Also add `group` module types.

Comment (Teja2045) [reply frontend/src/utils/authorizations.ts:282]: Which group module transactions can have Authz?
I could't find it in the comos-sdk repo

Comment (aleem1314) [reply frontend/src/utils/authorizations.ts:282]: https://github.com/cosmos/cosmos-sdk/blob/v0.46.13/proto/cosmos/group/v1/tx.proto#L16

Comment (Teja2045) [reply frontend/src/utils/authorizations.ts:282]: https://github.com/vitwit/resolute/blob/c7d36a1576a556477cd763f0f1b2e5f9f0925397/frontend/src/utils/authorizations.ts#L286C1-L288C8

Comment (Teja2045) [reply frontend/src/utils/authorizations.ts:282]: `if(authorizations[i].authorization?.msg.includes("cosmos.group.")) {
        result.daosEnabled = true;
      }`

File: frontend/src/features/authz/authzSlice.js [hunk 3]
```diff
@@ -451,19 +472,31 @@ export const authzSlice = createSlice({
       })
       .addCase(getGrantsToMe.fulfilled, (state, action) => {
         const chainID = action.payload?.chainID || "";
-        if (chainID.length) {
+        const { changeAuthzTab } = action.meta?.arg;
+
+        if (chainID?.length) {
+          let grants = action.payload.data.grants;
+          if (changeAuthzTab) {
```
Comment (aleem1314) [inline frontend/src/features/authz/authzSlice.js:479]: Do we need this condition? `changeAuthzTab` is always true.

**Review by aleem1314 — APPROVED** [review]
One question. Otherwise LGTM

File: frontend/src/features/authz/authzSlice.js (hunk 3 above)
Comment (Teja2045) [reply frontend/src/features/authz/authzSlice.js:479]: https://github.com/vitwit/resolute/blob/c7d36a1576a556477cd763f0f1b2e5f9f0925397/frontend/src/components/CustomAppBar.js#L59-L65C1

only the getGrantsToMe called in customAppBar.js have changeAuthzTab=true

The grantsToMe calls should't effect AuthzTabPanel

Comment (Teja2045) [reply frontend/src/features/authz/authzSlice.js:479]: Removing the condition shouldn't be a problem (though having it reduces the unnecessary AuthzTabPanel updates)

**Pull request merged**